src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2406,5 +2406,41 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::GetRadioRegion(uint16_t
     return error;
 }
 
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE
+template <typename InterfaceType, typename ProcessContextType>
+otError RadioSpinel<InterfaceType, ProcessContextType>::ConfigureEnhAckProbing(otLinkMetrics        aLinkMetrics,
+                                                                               const otShortAddress aShortAddress,
+                                                                               const otExtAddress & aExtAddress)
+{
+    otError error = OT_ERROR_NONE;
+    uint8_t flags = 0;
+
+    if (aLinkMetrics.mPduCount)
+    {
+        flags |= SPINEL_THREAD_LINK_METRIC_PDU_COUNT;
+    }
+
+    if (aLinkMetrics.mLqi)
+    {
+        flags |= SPINEL_THREAD_LINK_METRIC_LQI;
+    }
+
+    if (aLinkMetrics.mLinkMargin)
+    {
+        flags |= SPINEL_THREAD_LINK_METRIC_LINK_MARGIN;
+    }
+
+    if (aLinkMetrics.mRssi)
+    {
+        flags |= SPINEL_THREAD_LINK_METRIC_RSSI;
+    }
+
+    error = Set(SPINEL_PROP_RCP_ENH_ACK_PROBING, SPINEL_DATATYPE_UINT16_S, SPINEL_DATATYPE_EUI64_S,
+                SPINEL_DATATYPE_UINT8_S, aShortAddress, aExtAddress.m8, flags);
+
+    return error;
+}
+#endif
```

Comment:
I think we should almost never increase `SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION`. 
The newer Host software should always try to work with older RCP devices, except for when it's extremely difficult or inefficient.
Host software can always determine RCP capability according to its RCP version and issue commands accordingly. 

I am also suggesting that we add some checking (e.g. `static_assert(SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION == 4)`) to make sure SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION is **NOT** accidentally increased without thorough thinking and discussion.

Thoughts? @jwhui @abtink @konradderda

Reply:
Regarding the check, if the impact of changing `SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION` is so meaningful it could be set up in CI similarly to OT API version check: https://github.com/openthread/openthread/blob/main/.github/workflows/version.yml. I think that static_assert will be changed after a first compilation issue. 

Additionally, CI check could grep for the current and new number of `SPINEL_PROP_RCP_*` and enforce the incrementation `SPINEL_RCP_API_VERSION` or prevent the incrementation if it's not needed.

The one (potential) problem I can see with the current 'RCP vs Host' versioning system is that whenever you want to use a new RCP firmware you **must** update also the host side. I think it is not always necessary, as you may need the RCP version because of some important update etc., and if `SPINEL_RCP_API_VERSION` got updated because of properties that you know you do not need you are still forced to update the host side.